### ISSUE TITLE
Class is rendered instead of function if it has the same name

### ISSUE THREAD

**Issue by ianspektor** [mkdocstrings/python#76]
**Describe the bug**
I have a `src/operators/lag.py` file which defines a `lag()` function.
In `src/__init__.py`, I import the function: `from src.operators.lag import lag`.
When I use a `::: src.lag` identifier in my docs, the module's page gets rendered instead of the function.

**To Reproduce**
Create a project with the structure I described.

**Expected behavior**
Have the function's page rendered instead of its module's.

**Screenshots**
Here is an example of the unexpected behavior: https://temporian--154.org.readthedocs.build/en/154/reference/temporian/operators/lag/ (see that the module's docstring is rendered)
Here is the correct behavior (in this case the function and file have different naming): https://temporian--154.org.readthedocs.build/en/154/reference/temporian/operators/greater/ 

**System (please complete the following information):**
- `mkdocstrings-python` version: 0.21.2
- Python version: 3.10
- OS: happens in both MacOS and Linux

Thanks in advance for the help!

**Comment by pawamoy** [comment]
Hello, thanks for the report. Is your repository public, so I can take a look?

Griffe does suffer from a design flaw where objects of the same name can override modules in the objects tree, though in this case I'm not sure why it would happen, as you're importing the `lag` function into `src`, and not into `operators`. Do you happen to use wildcard imports (`from thing import *`)? They are very much not recommended, and can lead to some issues within Griffe, as they're quite hard to handle properly :sweat_smile:

**Comment by pawamoy** [comment]
You're welcome!

Wildcard imports will by default import *every object whose name doesn't start with an underscore*, whether it was defined in the module or imported from elsewhere. To prevent such re-exports and control what exactly is imported by wildcard imports, you can define the variable `__all__` in your module, which is a list of object names (strings).

**Comment by ianspektor** [comment]
Yep, I tried that - but what about nested * imports? If module C defines its __all__, then B imports * from C, and then I need A to import * from B, how do I define B's __all__? I would need it to extend C's (I found some [hacks](https://stackoverflow.com/questions/18957205/extending-all-how-can-i-take-module-all-list) to  but decided my separate file solution was both less verbose and less repetitive).

**Comment by pawamoy** [comment]
Some devs import the `__all__` variable from C to extend the one from B:

```python
from pkg.c import *
from pkg.c import __all__ as c_all

__all__ = ["b_thing", "b_other_thing"] + c_all
```

...though not all operations are supported by Griffe yet (there are many ways to combine lists).
